frontend/src/metabase/public/components/widgets/AdvancedSettingsPane.jsx

```diff
@@ -52,6 +52,9 @@ const AdvancedSettingsPane = ({
           className="pt1"
           displayOptions={displayOptions}
           onChangeDisplayOptions={onChangeDisplayOptions}
+          // We only show the "Download Data" toggle if the users are pro/enterprise
+          // and they're sharing a question metabase#23477
+          showDownloadDataButtonVisibilityToggle={resourceType === "question"}
```

Comment:
[We only show the download button on embedded questions](https://metaboat.slack.com/archives/CKZEMT1MJ/p1655908678259929?thread_ts=1655856289.399469&cid=CKZEMT1MJ).